src/snowflake/connector/local_storage_client.py

```diff
@@ -26,42 +26,45 @@ def __init__(
         meta: SnowflakeFileMeta,
         stage_info: dict[str, Any],
         chunk_size: int,
-        use_s3_regional_url: bool = False,
     ) -> None:
         super().__init__(meta, stage_info, chunk_size)
         self.data_file = meta.src_file_name
         self.full_dst_file_name: str = os.path.join(
             stage_info["location"], os.path.basename(meta.dst_file_name)
         )
+        if meta.local_location:
+            self.stage_file_name = self.full_dst_file_name
+            self.full_dst_file_name = os.path.join(
+                meta.local_location, meta.dst_file_name
+            )
 
-    def get_file_header(self, filename: str) -> None:
+    def get_file_header(self, filename: str) -> FileHeader | None:
         """
         Notes:
             Checks whether the file exits in specified directory, does not return FileHeader
         """
-        # TODO return a FileHeader sometime
-        target_dir = os.path.join(
-            os.path.expanduser(self.stage_info["location"]),
-            filename,
-        )
-        if os.path.isfile(target_dir):
-            self.meta.result_status = ResultStatus.UPLOADED
-        else:
-            self.meta.result_status = ResultStatus.NOT_FOUND_FILE
+        if os.path.isfile(filename):
+            return FileHeader(None, os.stat(filename).st_size, None)
+        return None
 
     def download_chunk(self, chunk_id: int) -> None:
-        with open(self.full_dst_file_name, "rb") as sfd:
+        with open(self.stage_file_name, "rb") as sfd:
             with open(
                 os.path.join(
-                    self.meta.local_location, os.path.basename(self.meta.dst_file_name)
+                    self.meta.local_location,
+                    os.path.basename(self.intermediate_dst_path),
                 ),
-                "wb",
+                "rb+",
             ) as tfd:
-                tfd.seek(chunk_id * self.chunk_size)
-                sfd.seek(chunk_id * self.chunk_size)
-                tfd.write(sfd.read(self.chunk_size))
+                if self.num_of_chunks == 1:
```

Comment:
🤔 Why do we need a special case here?

Reply:
Ha this is kinda interesting because `num_of_chunks == 1` could mean: 

- We use chunked upload/download and there is only one chunk
- We don't use chunked upload/download

The second case is a bit tricky here because if we decided to not use chunked upload/download because of `multipart_threshold`, and the total data size is bigger than one chunk, then we will miss data.